nvdaHelper/local/wasapi.cpp

```
@@ -610,6 +606,12 @@ HRESULT WasapiPlayer::stop() {
 }
 
 void WasapiPlayer::completeStop() {
+	HRESULT hr = client->Reset();
+	if (FAILED(hr)) {
+		LOG_ERROR(L"Couldn't reset stream: " << hr);
```

We may want to make this `LOG_DEBUGWARNING` eventually if this is regularly caused by audio devices being unplugged or whatever. However, I think it makes sense for it to be ERROR at least initially so we don't swallow/understate what might turn out to be real problems.

---

Having this as LOG_ERROR for me causes some kind of freeze / deadlock when the original bug occurs. Here is a fragment of my log of the secure copy as  it connects:
```
INFO - _remoteClient.session.RemoteSession.handleClientConnected (07:40:14.275) - MainThread (20852):
Client connected: {'id': 62, 'connection_type': 'master'}
INFO - watchdog.waitForFreezeRecovery (07:40:14.777) - watchdog (17848):
Starting freeze recovery after 0.5002936000018963 seconds.
DEBUGWARNING - watchdog.waitForFreezeRecovery (07:40:14.777) - watchdog (17848):
Listing stacks for Python threads:
Python stack for thread 19936 (nvwave.playWaveFile(controlled.wav)):
  File "threading.pyc", line 1002, in _bootstrap
  File "threading.pyc", line 1045, in _bootstrap_inner
  File "threading.pyc", line 982, in run
  File "nvwave.pyc", line 128, in play
  File "nvwave.pyc", line 337, in feed
  File "NVDAHelper.pyc", line 323, in nvdaControllerInternal_logMessage
  File "logHandler.pyc", line 262, in _log
  File "logging\__init__.pyc", line 1634, in _log
  File "logging\__init__.pyc", line 1644, in handle
  File "logging\__init__.pyc", line 1706, in callHandlers
  File "logHandler.pyc", line 398, in handle
  File "extensionPoints\__init__.pyc", line 56, in notify
  File "extensionPoints\util.pyc", line 230, in callWithSupportedKwargs
  File "nvwave.pyc", line 566, in playErrorSound
  File "nvwave.pyc", line 112, in playWaveFile
  File "nvwave.pyc", line 387, in stop
  File "nvwave.pyc", line 459, in _setVolumeFromConfig
  File "nvwave.pyc", line 428, in setVolume

Python stack for thread 17848 (watchdog):
  File "threading.pyc", line 1002, in _bootstrap
  File "threading.pyc", line 1045, in _bootstrap_inner
  File "threading.pyc", line 982, in run
  File "watchdog.pyc", line 163, in _watcher
  File "watchdog.pyc", line 170, in waitForFreezeRecovery
  File "logHandler.pyc", line 64, in getFormattedStacksForAllThreads

Python stack for thread 1124 (queue_thread):
  File "threading.pyc", line 1002, in _bootstrap
  File "threading.pyc", line 1045, in _bootstrap_inner
  File "threading.pyc", line 982, in run
  File "_remoteClient\transport.pyc", line 540, in sendQueue
  File "queue.pyc", line 171, in get
  File "threading.pyc", line 327, in wait

Python stack for thread 22192 (Thread-4_connector_loop):
  File "threading.pyc", line 1002, in _bootstrap
  File "threading.pyc", line 1045, in _bootstrap_inner
  File "_remoteClient\transport.pyc", line 705, in run
  File "_remoteClient\transport.pyc", line 352, in run
  File "_remoteClient\transport.pyc", line 390, in _readLoop

Python stack for thread 1604 (winInputHook):
  File "threading.pyc", line 1002, in _bootstrap
  File "threading.pyc", line 1045, in _bootstrap_inner
  File "threading.pyc", line 982, in run
  File "winInputHook.pyc", line 108, in hookThreadFunc

Python stack for thread 16096 (UIAHandler.UIAHandler.MTAThread):
  File "threading.pyc", line 1002, in _bootstrap
  File "threading.pyc", line 1045, in _bootstrap_inner
  File "threading.pyc", line 982, in run
  File "UIAHandler\__init__.pyc", line 554, in MTAThreadFunc
  File "queue.pyc", line 171, in get
  File "threading.pyc", line 327, in wait

Python stack for thread 2420 (synthDrivers._espeak.BgThread):
  File "threading.pyc", line 1002, in _bootstrap
  File "threading.pyc", line 1045, in _bootstrap_inner
  File "synthDrivers\_espeak.pyc", line 207, in run
  File "queue.pyc", line 171, in get
  File "threading.pyc", line 327, in wait

Python stack for thread 21160 (hwIo.ioThread.IoThread):
  File "threading.pyc", line 1002, in _bootstrap
  File "threading.pyc", line 1045, in _bootstrap_inner
  File "hwIo\ioThread.pyc", line 264, in run

Python stack for thread 2132 (ScheduleThread):
  File "threading.pyc", line 1002, in _bootstrap
  File "threading.pyc", line 1045, in _bootstrap_inner
  File "utils\schedule.pyc", line 90, in run

Python stack for thread 20852 (MainThread):
  File "nvda.pyw", line 309, in <module>
  File "core.pyc", line 1049, in main
  File "wx\core.pyc", line 2262, in MainLoop
  File "wx\core.pyc", line 3427, in <lambda>
  File "extensionPoints\__init__.pyc", line 56, in notify
  File "extensionPoints\util.pyc", line 230, in callWithSupportedKwargs
  File "_remoteClient\session.pyc", line 376, in handleChannelJoined
  File "_remoteClient\session.pyc", line 359, in handleClientConnected
  File "_remoteClient\session.pyc", line 224, in handleClientConnected
  File "_remoteClient\cues.pyc", line 77, in clientConnected
  File "_remoteClient\cues.pyc", line 57, in _playCue
  File "nvwave.pyc", line 112, in playWaveFile
  File "nvwave.pyc", line 387, in stop
```

---

Note that I hear the start of the connection sound and then it is cut off. And then just silence. No speech on either machine.

---

Hmm. I guess error isn't ideal precisely because it plays a sound and we're potentially dealing with an error playing a sound. That said, you ideally shouldn't be hitting this error any more at all, so that's a bit worrying. Can you try with LOG_DEBUGWARNING instead? I'd like to know what the error code is; i.e. whether it's still AUDCLNT_E_BUFFER_OPERATION_PENDING or something else.